Source/msg.cpp

```diff
@@ -1520,26 +1530,32 @@ DWORD OnSendPlayerInfo(const TCmd *pCmd, int pnum)
 
 DWORD OnPlayerJoinLevel(const TCmd *pCmd, int pnum)
 {
-	auto *p = (TCmdLocParam1 *)pCmd;
+	const auto &message = *reinterpret_cast<const TCmdLocParam1 *>(pCmd);
+	const Point position { message.x, message.y };
 
 	if (gbBufferMsgs == 1) {
-		SendPacket(pnum, p, sizeof(*p));
-		return sizeof(*p);
+		SendPacket(pnum, &message, sizeof(message));
+		return sizeof(message);
+	}
+
+	int playerLevel = message.wParam1;
+	if (playerLevel >= MAXCHARLEVEL || !InDungeonBounds(position)) {

```

Comment:
```suggestion
	if (playerLevel > (gbIsHellfire ? 24 : 16) || !InDungeonBounds(position)) {
```

Overlooked this in the first review. 😝 
`plrlevel` corresponds to dungeon `currlevel`.
`_pLevel` is the character level.

Reply:
Fixed